### DISCUSSION TITLE
What is the best way to initialize class fields ?

### DISCUSSION THREAD

**Discussion by andr1972** [pfusik/cito#152] [Q&A]
Since constructors have not parameters, so need call method like 
```
public class Diff
{
    string a; 
    string b;
    public void Init!(string a, string b) 
   {
        this.a = a;
        this.b = b;
    }

   public void Compare()
   {
   }
}

public class Program
{
    public static void Main(string[] args)
    {
            Diff() diff;
            diff.Init("ABCABBA", "CBABAC");
            diff.Compare();
    }
}
```
, best way is call method Init or other convention?

**Comment by pfusik** [comment]
From Fusion code you can use:

    Diff() diff = { a = "ABCABBA", b = "CBABAC" };

Make the fields `internal` if you use the above from a different class.

Instead of calling the method `Init`, you can use a more descriptive name, such as `SetAB`. Or maybe you should split it into two setter methods `SetA` and `SetB` ?

Also, why do you store the string references in the object? You might pass them to the `Compare` method.
